### DISCUSSION TITLE
An 'Invalid' boundary 'for' Multipart /form-data 'request' error occurred regarding the use of 'multipart'

### DISCUSSION THREAD

**Discussion by Sparkel-wsl** [tokio-rs/axum#1765] [Q&A]
- [ ] I have looked for existing issues (including closed) about this

### Version
cargo 1.64.0 (387270bc7 2022-09-16)

### Platform
Windows 10 64-bit

### Crates
axum = { version = "0.6.1",features=["multipart"]}

### Description

Hello, I wrote the backend of 'ueditor' in 'rust', normal images can be uploaded normally, http headers are: Content-type:application/json, but when uploads the doodle file, we get an error 'multipart/form-data' request '. When I look at the http request headers, I see Content-Type: application/x-www-form-urlencoded, which is a base64 string in form-data. Looking at the axum documentation, I found that axum::Form accepts.

But uploading images and doodling must be the same request address. How can axum be compatible with both headers Content-type:application/json Content-Type: application/x-www-form-urlencoded request.

你好，我使用 `rust` 编写 `ueditor` 的后端，普通图片可以正常上传，http headers是：Content-type:application/json，但在上传涂鸦文件时，发生了这么一个错误`multipart/form-data` request`。我查看http request headers 发现 Content-Type : application/x-www-form-urlencoded，在 form-data 里面的值是 base64字符串。我查阅 axum的文档后发现，可以用axum::Form 接收。

但是上传图片和涂鸦必须是一个请求地址，axum 如何同时兼容headers Content-type:application/json Content-Type : application/x-www-form-urlencoded 的请求。

![image](https://user-images.githubusercontent.com/103182571/219606521-8f70b338-159e-4e4f-8591-d2ba62ea7384.png)

**Comment by davidpdrsn** [comment]
Have a look at this example https://github.com/tokio-rs/axum/blob/main/examples/parse-body-based-on-content-type/src/main.rs

**Reply by Sparkel-wsl** [reply]
Thank you very much

**Comment by iyra** [comment]
I ran into this today; the example suggested in the other comment may not be necessary.

To anyone else coming across this, ensure that your form element (or your Postman request) is formatted using "multipart/form-data" (i.e. `<form action="..." method="post" enctype="multipart/form-data">.`)

The default of x-www-urlencoded does not play nicely with Axum's multipart, nor with axum_typed_multipart.
